rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,345 @@
+
+#### Consequences
+
+* If we already have assets in various ranges, the runtime migration process
+will become more complicated as we translate them to existing in one range.
+  
+### Remove Asset ID Reservation
+
+> We do not need [asset ID reservation]. We just runtime config assets to be 
+permissioned. When we will have permisionless assets they may already have 
+reservation (any way making fork with reservation is easy later).
+
+Asset reservation helps us automate new asset IDs. However, for the scope of 
+release two - this may not be necessary. We could instead hard-code asset IDs 
+for LP tokens and other instances.
+
+Given we can currently manage without automated asset creation, as we can forgo
+the asset creation requirements for release two and ensure that all asset data
+is still made available.
+
+#### Consequences
+
+* Minor runtime migration will be needed for deleting Currency Factory's 
+storage.
+
+### Use an Externally Maintained Asset ID Reservation System
+
+Neither option presented seems to handle the reserving of asset IDs. Therefore, 
+neither of these will fully replace currency factory and would more correctly 
+replace our own pallet-assets. A replacement for pallet-assets may be a good 
+idea but would require much more substantial changes.
+
+* **Parity's pallet-assets**
+  
+  Forking this pallet and maintaining extra features we need (hold) is an 
+  option.
+  
+  > It is not clear from RFC why substrate assets are good. IMHO it should be 
+  clarified - year of man work to make UI/QA/FE/users happy is what we lack. 
+  IMHO better to have fork of assets and collaborate with ecosystem to tune 
+  wallets. Do not write own wallets.
+  
+  #### Pros
+  
+  * Very nice UI compatibility with Polkadot Dashboard
+
+  * Maintained by Parity
+
+  * Simple Interface
+
+  * Can run two pallets at same time for two purposes (external and mintable) 
+  (Demonstrated by Moonbeam)
+
+  * Handles WASM contracts addresses as IDs
+
+  * Permissionless when needed
+
+  * Supported by Polkadot wallets
+  
+  * Supports asset metadata
+  
+  #### Cons
+  
+  * No asset ID reservation system, only prevents duplicates
+
+  * Hold functionality is not present (We depend on this in several cases)
+
+* **Moonbeam's pallet-asset-manager**
+
+Note: Not looking at using or forking this directly, but instead how they use 
+different instances of the same pallet for mintable and external assets.
+
+  #### Pros
+  
+  * Maintained by moonbeam
+  
+  #### Cons
+  
+  * Partially specialized
+
+  * No asset ID reservation system, only prevents duplicates
+
+#### Consequences
+
+* Any of these options will require a difficult runtime migration as we move 
+existing storage for pallet-assets/pallet-asset-registry/pallet-currency-factory 
+to an alternative.
+
+## Conclusion
+
+After analyzing possible solutions, the best course of action to deprecate 
+Currency Factory is to maintain a fork of Parity's pallet-assets, embrace the 
```

Comment:
Note that pallet-assets is still a work in progress https://github.com/paritytech/substrate/issues/8453 , we need `fungibles::*Hold` traits for staking in release 3, which pallet-assets does not support. Any solution suggesting adoption of that pallet need to take into account orml-tokens features that we need for release 3. I would rather deal with a migration into pallet-assets(it supports that for genesis https://github.com/paritytech/substrate/blob/master/frame/assets/src/lib.rs#L362) later than delay our feature roadmap. Note that the number of tokens(LPT * 3, PBLO, pool shares * 3) would not be that large even after release 3 and staking. We can even hard code a migration for that. 

I propose the following steps to impl in brief

#### Release 2: 
1. Make sure that we don't use currency-factory and create and track(1) the assets in assets-registry with `MultiLocation` even for local assets.
2. Allow Pablo to create assets(LPT) through assets-registry (avoid use of currency-factory)
3. Make sure one can not use local assets to pay gas fees in XCM transfers (Which seems to be the case AFAIK)
4. For ED for LPT: Asset ratios gets defined in a suitable way at the time of creation for LP (1 LP = value of provided liquidity in PICA / number of tokens)

#### Release 3: 
1. Introduce staking with pool share tokens (Just for context, not related to RFC)
2. Remove currency-factory all together (Just to minimize the changes for release 2)
3. Possibly upgrade substrate to latest production version (assess later).

#### Release 4 or later
1. Assess/Introduce frame/pallet-assets with a migration for existing set of assets if `fungibles::*Hold` trait impls are available.
2. Remove use of our own pallet-assets together with orml-tokens use.